File: Replace-Words/RW.go
package main

import (
	. "fmt"
	. "strings"
)

func replaceWords(dict []string, sentence string) string {
	sentlist := Split(sentence, " ")
	for _, key := range dict {
		tempList := []string{}
		for _, word := range sentlist {
			if len(key) <= len(word) && key == word[:len(key)] {
				tempList = append(tempList, key)
			} else {
				tempList = append(tempList, word)
			}
		}
		sentlist = tempList
	}
	Println(sentlist)

	result := ""
	for _, word := range sentlist {
		result += word + " "
	}

	return result
}

func main() {
	Println(replaceWords([]string{"cat", "bat", "rat"}, "the cattle was rattled by the battery"))
}

File: Replace-Words/RW.scala
def replaceWords(dict:Array[String], sentence:String):String = {
  var tempList = Array[String]()
  var sentList = sentence.split(" ")
  for (keyword <- dict) {
    tempList = Array()
    for (word <- sentList) {
      if ((keyword.length <= word.length) && (keyword == word.slice(0,keyword.length))){
        tempList = tempList :+ keyword
      }else {
        tempList = tempList :+ word
      }
    }
    sentList = tempList
  }
  sentList.mkString(" ")
}

println(replaceWords(Array("cat", "bat", "rat"), "the cattle was rattled by the battery"))
